ng2-components/ng2-activiti-diagrams/src/components/diagram.component.ts

```diff
@@ -71,14 +74,35 @@ export class DiagramComponent {
     ngOnChanges(changes: SimpleChanges) {
         this.reset();
         this.diagramColorService.setTotalColors(this.metricColor);
-        this.getProcessDefinitionModel(this.processDefinitionId);
+        if (this.processDefinitionId) {
+            this.getProcessDefinitionModel(this.processDefinitionId);
+        } else {
+            this.getRunningProcessDefinitionModel(this.processInstanceId);
+        }
+    }
+
+    getRunningProcessDefinitionModel(processInstanceId: string) {
+        this.diagramsService.getRunningProcessDefinitionModel(processInstanceId).subscribe(
+            (res: any) => {
+                this.diagram = new DiagramModel(res);
+                this.raphaelService.setting(this.diagram.diagramWidth + this.PADDING_WIDTH,
+                                            this.diagram.diagramHeight + this.PADDING_HEIGHT);
+                this.setMetricValueToDiagramElement(this.diagram, this.metricPercentages, this.metricType);
+                this.onSuccess.emit(res);
+            },
+            (err: any) => {
+                this.onError.emit(err);
+                this.logService.error(err);
```

Comment:
The log error is already in the catch of the service . In this way we are going to log twice. Please remove it from here and leave only the emit